README.md

```diff
@@ -1,28 +1,84 @@
+
+Given that we have capacity of 20K RU per second, we know that we could complete the work in 100 seconds if we could spread it out. 
+
+However, each process might try and send their own 100K records in parallel and with no knowledge of each other. This would cause Cosmos to start issuing `429 TooManyRequests` error messages and given the volume it might even cut you off with `503 ServiceUnavailable` error messages. 
+
+Batcher solves this problem by allowing you to share the capacity across multiple replicas and controlling the flow of traffic so you don't exceed the 20K RU per second.
+
+### Reliability - rate limiting feature
+
+Consider this scenario:
+- You have a datastore (SQL Server for example) that does not have a native rate limiting feature. 
+- Just because you can send to your datastore as many requests as you want in a timeframe, that does not mean it can process those requests efficiently or without error.
+
+Batcher allows you to apply a rate limiter.
+
+### Scalability - calibrate capacity against operation time
+
+Consider this scenario:
+- You have an Azure Cosmos database 
+- You need to provision it with fixed capacity as this allows you to select the cost you are willing to pay for your database. 
+
+Batcher will ensure that you don't exceed this capacity by lengthening the time it takes for your operations to complete. Therefore, if you find that your application takes too long for operations to complete, you can increase the capacity. If you want to save money, you can decrease the capacity.
+
+## Batcher Components
+
+- __Batcher__: A Batcher is created for each datastore that has capacity you wish to respect. Batchers are long-lived.
+
+- __Watcher__: A Watcher is created for each process you wish to manage. Lots of Watchers will share the same Batcher. The Watcher receives the batches as they become available. Watchers are short-lived. For instance, if your solution is an HTTP server, you will probably create a Watcher with each request, send your Operations to a shared Batcher, and then get batches for processing back on your Watcher. If you need to handle different types of Operations that are processed in different ways or if they have different characteristics (such as an optimal batchsize), you might create a separate Watcher for each of those use cases.
+
+- __Operation__: Operations are enqueued into the Batcher. An Operation has an associated "Watcher", a "cost", a designation of whether or not it can be batched, a counter for the number of times it has been "attempted", and a "payload" (which can be anything you want).
 
 There are also 2 rate limiters provided out-of-the-box...
 
 - __ProvisionedResource__: This is a simple rate limiter that has a fixed capacity per second.
 
 - __AzureSharedResource__: This rate limiter allows you to reserve a fixed amount of capacity and then share a fixed amount of capacity across multiple processes.
```

Comment:
TODO: As this is the first reference to AzureSharedResource, clarify Azure Storage is used for this and why.

Reply:
added note.